### PR TITLE
[ppo critic] fix EOS token value to zero

### PR DESCRIPTION
**Pull request by Jackory**
### Checklist Before Starting

- [x] Search for similar PR(s).

### What does this PR do?

> Add one-line overview of what this PR aims to achieve or accomplish. 

 For PPO critic training, the value of EOS tokens should be zero and should not be fitted. However, the current implementation does not mask the EOS token values, resulting in non-zero EOS token values. Although the learning target is zero, when PPO GAE lambda < 1, this affects the advantage calculation for tokens preceding EOS, thereby impacting performance.

### High-Level Design

> Demonstrate the high-level design if this PR is complex.

### Specific Changes

> List the specific changes.

### API

> Demonstrate how the API changes if any.

### Usage Example

> Provide usage example(s) for easier usage.

```python
# Add code snippet or script demonstrating how to use this 
```

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluatuion results, etc.

### Additional Info.

- **Issue Number**: Fixes issue # or discussion # if any.
- **Training**: [Note which backend this PR will affect: FSDP, Megatron, both, or none]
- **Inference**: [Note which backend this PR will affect: vLLM, SGLang, both, or none]

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl?tab=readme-ov-file#contribution-guide).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl?tab=readme-ov-file#code-linting-and-formatting).
- [x] Add `[BREAKING]` to the PR title if it breaks any API.
- [x] Update the documentation about your changes in the [docs](https://github.com/volcengine/verl/tree/main/docs).
- [x] New CI unit test(s) are added to cover the code path.
- [x] Rely on existing unit tests on CI that covers the code path.

### CODE REVIEW

File: verl/workers/critic/dp_critic.py [hunk 1]
```diff
@@ -163,16 +168,17 @@ def compute_values(self, data: DataProto) -> torch.Tensor:
...
             indices = list(itertools.chain.from_iterable(indices))
             assert len(indices) == values.size(0), f"{len(indices)} vs. {values.size()}"
             revert_indices = torch.tensor(get_reverse_idx(indices), dtype=torch.long)
             values = values[revert_indices]
-        values = values * attention_mask[:, -response_length - 1 : -1]
+            
+        responses = data.batch["responses"]
+        attention_mask = data.batch["attention_mask"]
+        response_length = responses.size(1)
+        values = values * attention_mask[:, -response_length:]
```
Comment (tongyx361) [inline verl/workers/critic/dp_critic.py]:
```suggestion
        response_mask = attention_mask[:, -response_length:]
        values = values * response_mask # Only action tokens have values
```

File: verl/workers/critic/megatron_critic.py [hunk 2]
```diff
@@ -117,8 +118,9 @@ def compute_values(self, data: DataProto) -> DataProto:
                 values = torch.empty_like(attention_mask, dtype=torch.float32)
 
             # each tp ranks should contain the same value
-            values = values * attention_mask
+            response_mask = attention_mask[:, -response_length:]
             values = values[:, -response_length - 1 : -1]
+            values = values * response_mask
```
Comment (tongyx361) [inline verl/workers/critic/megatron_critic.py]:
```suggestion
            values = values[:, -response_length - 1 : -1] # Values are predicted at the ends of prefixes, e.g., the last prompt token
            response_mask = attention_mask[:, -response_length:]
            values = values * response_mask # Only action tokens have values
```

**Review by tongyx361 — APPROVED** [review]

**Comment by tongyx361** [comment]
Since commit 59ceba6 has passed the CI, we skip the CI after commits that don't affect the logic to save resources.
